src/sql-pretty/src/lib.rs

```diff
@@ -22,39 +23,69 @@ use crate::doc::{
 
 pub use crate::doc::doc_expr;
 
+pub const DEFAULT_WIDTH: usize = 100;
+
 const TAB: isize = 4;
 
-fn to_doc<T: AstInfo>(v: &Statement<T>) -> RcDoc {
+fn to_doc<T: AstInfo>(v: &Statement<T>, config: PrettyConfig) -> RcDoc {
     match v {
-        Statement::Select(v) => doc_select_statement(v),
-        Statement::Insert(v) => doc_insert(v),
-        Statement::CreateView(v) => doc_create_view(v),
-        Statement::CreateMaterializedView(v) => doc_create_materialized_view(v),
-        Statement::Copy(v) => doc_copy(v),
-        Statement::Subscribe(v) => doc_subscribe(v),
-        Statement::CreateSource(v) => doc_create_source(v),
-        _ => doc_display(v, "statement"),
+        Statement::Select(v) => doc_select_statement(v, config),
+        Statement::Insert(v) => doc_insert(v, config),
+        Statement::CreateView(v) => doc_create_view(v, config),
+        Statement::CreateMaterializedView(v) => doc_create_materialized_view(v, config),
+        Statement::Copy(v) => doc_copy(v, config),
+        Statement::Subscribe(v) => doc_subscribe(v, config),
+        Statement::CreateSource(v) => doc_create_source(v, config),
+        _ => doc_display(v, config, "statement"),
     }
 }
 
+#[derive(Clone, Copy)]
+pub struct PrettyConfig {
+    pub width: usize,
+    pub format_mode: FormatMode,
+}
+
 /// Pretty prints a statement at a width.
-pub fn to_pretty<T: AstInfo>(stmt: &Statement<T>, width: usize) -> String {
-    format!("{};", to_doc(stmt).pretty(width))
+pub fn to_pretty<T: AstInfo>(stmt: &Statement<T>, config: PrettyConfig) -> String {
+    format!("{};", to_doc(stmt, config).pretty(config.width))
```

Comment:
Threading a `PrettyConfig` into the `doc*` functions doesn't seem right?

Reply:
See discussion [here](https://github.com/MaterializeInc/materialize/pull/31933#discussion_r2017283330).